build/config/compiler/compiler.gni

```diff
@@ -44,7 +44,7 @@ declare_args() {
   # If true, optimize for size.
   # Default to favoring speed over size for platforms not listed below.
   optimize_for_size =
-      !is_high_end_android && (is_android || is_ios || is_castos)
+      !(is_high_end_android || is_starboard) && (is_android || is_ios || is_castos)
```

Comment:
It looks to me like https://github.com/youtube/cobalt/pull/2969, by adding `//build/config/compiler:default_optimization` to `default_compiler_configs` in https://github.com/youtube/cobalt/blob/main/starboard/build/config/BUILDCONFIG.gn, made  `optimize_for_size` more important in Cobalt builds. Things are a little confusing now because some optimization flags get added by `//starboard/build/config:size` (https://github.com/youtube/cobalt/blob/main/starboard/build/config/BUILDCONFIG.gn#L164), and then possibly different optimization flags get added to the compilation line by `//build/config/compiler:default_optimization` (based on the value of `optimize_for_size`).

It would be good to clean this up a bit in a separate change. @andrewsavage1 could you possibly take a look as part of b/330781200?

Reply:
I opened b/350525014 to discuss optimizing for size again, so maybe we can just discuss this as part of that.